pkg/relayer/session/proof.go

```diff
@@ -114,15 +114,22 @@ func (rs *relayerSessionsManager) waitForEarliestSubmitProofsHeightAndGeneratePr
 	proofsWindowOpenBlock := rs.waitForBlock(ctx, proofWindowOpenHeight)
 	// TODO_MAINNET: If a relayminer is cold-started with persisted but unproven ("late")
```

Comment:
```go
// TODO_MAINNET: Handle cold-start edge case for "late" sessions.
//
// Problem:
// - If RelayMiner starts with persisted but unproven ("late") sessions,
//    the corresponding proofsWindowOpenBlock may never be observed.
// - A "late" session is:
//     • Unclaimed
//     • Earliest commit height has already passed (i.e., it's in the past)
// - In this scenario, the block observable won't emit the expected signal.
//
// Planned fix:
// - Use a block query client to retroactively populate the observable
//   during block client construction.
//
// Note:
// - Once this is implemented, this failure check and branch can be removed.
```

Reply:
This comment is no longer applicable. Cold starts are already handled.